app/controllers/results_controller.rb

```diff
@@ -138,12 +138,11 @@ def edit
 
   def run_tests
     grouping_id = params[:grouping_id]
-    # TODO: The submission id is set incorrectly as the grouping_id
     submission_id = Result.find(params[:id]).submission.id
 
     begin
       AutomatedTestsHelper.request_a_test_run(grouping_id,
-                                              'submission',
+                                              'request',
```

Comment:
True, but shouldn't it be exactly that way since it is triggered here by a student pressing the "run tests" button based on tokens?